packages/react-server/src/ReactFlightServer.js

```diff
@@ -111,10 +110,13 @@ export function createRequest(
   return request;
 }
 
-function attemptResolveElement(element: React$Element<any>): ReactModel {
-  const type = element.type;
-  const props = element.props;
-  if (element.ref !== null && element.ref !== undefined) {
+function attemptResolveElement(
+  type: any,
+  key: null | React$Key,
+  ref: mixed,
+  props: any,
+): ReactModel {
```

Comment:
Can you explain what you fixed here? I don't get it.

Reply:
It's a separate commit https://github.com/facebook/react/pull/20171/commits/29b9646b131ade8979e11b100b24ac2f8406de0c

Reply:
Yeah but I don't understand what it was fixing

Reply:
Ah I see the `REACT_MEMO_TYPE` case. Was it losing `element.key` and `element.ref` earlier because you did `createElement(type, element.props)`?

Reply:
You did. :) But yea and we might remove the export at some point too. Even so, it's just less efficient.